### PR TITLE
build: add Angular 19 support

### PR DESCRIPTION
**Pull request by arturovt**
Issue: #2268

### CODE REVIEW

**Review by rhutchison — APPROVED** [review]

File: .circleci/config.yml
Comment (markwhitfeld) [inline .circleci/config.yml:181]: This job needs to be added to the `build-workflow`.

**Review by markwhitfeld — CHANGES_REQUESTED** [review]
Great work, but it seems that in the following file:
`packages\store\experimental\src\pending-tasks.ts`
We import `ExperimentalPendingTasks` from `@angular/core`.
In Angular 19, this is now available as `PendingTasks` from `@angular/core`.

This means that we can move the `withExperimentalNgxsPendingTasks` utility from the `@ngxs/store/experimental` sub package to `withNgxsPendingTasks` in the main `@ngxs/store` package (unless you can think of a more ideal place?).
Related to this change would be a version bump of NGXS v18 to NGXS v19 (especially because of this breaking change)🎉

**Comment by arturovt** [comment]
I agree.

**Comment by arturovt** [comment]
But we should wait until the Nx allows migrating to Angular 19 in order to update the workspace version too.

**Comment by markwhitfeld** [comment]
What is the dependency on NX which would be blocking us?
Can we not release an NGXS v19 independently of the version of NX?
Surely we can do the workspace update separately.

PS. We should still keep the experimental subpackage around. 
We can export the `withExperimentalNgxsPendingTasks` with a deprecation notice. This function would just use the `withNgxsPendingTasks` function that has moved to the main package.

**Comment by arturovt** [comment]
We can but not with the pending tasks change. The workspace is using Angular 18 which doesn't expose `PendingTasks` from the core.

**Comment by markwhitfeld** [comment]
Ok, so we are blocked until this NX PR is merged:
https://github.com/nrwl/nx/pull/28847
That NX PR is, in turn blocked by a number of their dependencies, waiting for upgrade.

Is there any way that we can decouple the angular version that our libraries reference from the version of NX?
I'm wondering if there is something we can do with overrides in the `package.json`, or if we could use the aliasing functionality of tsconfigs to "redirect" the referenced angular version use in our libraries?
What we build with doesn't necessarily determine what our code references. Any ideas?

**Comment by arturovt** [comment]
> We can export the withExperimentalNgxsPendingTasks with a deprecation notice.

This would throw an error that ` Entry point @ngxs/store has a circular dependency on itself.`

**Comment by markwhitfeld** [comment]
The experimental subpackage already references things from the
***@***.***/store` library.
We would be moving the functionality to the core lib and referencing it
there (having the same direction of reference as the existing refs).

**Comment by arturovt** [comment]
I have looked at what NgRx did, they upgraded workspace version w/o waiting for the Nx upgrade.

I upgraded the workspace version and had to update unit tests because components are standalone by default now.

**Comment by markwhitfeld** [comment]
Great work!
There may be some usefulness to still have some components as `standalone:
false` where we are testing things about modules, but I think that most can
be converted.

**Comment by arturovt** [comment]
@markwhitfeld what do we with ng16/17/18 integrations? They're not compatible anymore because we're using `PendingTasks`. Do we remove them completely because we want to bump the version to `ngxs@19` and make it compatible with `angular >= 19.0.0`?

**Comment by tejasv-tagline** [comment]
Is anyone taking care of changes for this PR? Many users are waiting for Angular 19 support.

File: packages/store/experimental/src/pending-tasks.ts
Comment (markwhitfeld) [inline packages/store/experimental/src/pending-tasks.ts:1]:
```suggestion
import { withNgxsPendingTasks } from '@ngxs/store';

/**
 * Required for correct functioning of SSR apps. 
 *
 * @deprecated
 * This experimental export is deprecated in favour of the non-experimental export. This is no longer an experimental feature, but is now available as `withNgxsPendingTasks` in the main `@ngxs/store` package.
 */
export function withExperimentalNgxsPendingTasks () {
  return withNgxsPendingTasks(); 
}
```

File: packages/store/src/pending-tasks.ts
Comment (markwhitfeld) [inline packages/store/src/pending-tasks.ts:7]:
```suggestion
 * This feature that contributes to app stability, which is required during
```

File: packages/store/src/pending-tasks.ts
Comment (markwhitfeld) [inline packages/store/src/pending-tasks.ts:11]:
```suggestion
  * Including `withNgxsPendingTasks` in your `provideStore` for your SSR app will resolve the above issue.
  */
```

File: packages/store/experimental/tests/preboot-wait-stable.spec.ts
Comment (markwhitfeld) [inline packages/store/experimental/tests/preboot-wait-stable.spec.ts:1]: We should add this test to the main package's tests.

**Review by markwhitfeld — CHANGES_REQUESTED** [review]

**Review by markwhitfeld — APPROVED** [review]
Great work!!!!
Let's get this released!

**Pull request merged**
